packages/grid/_modules_/grid/hooks/features/useColumnReorder.tsx

```diff
@@ -0,0 +1,150 @@
+import * as React from 'react';
+import { ColDef } from '../../models/colDef';
+import { useLogger } from '../utils';
+import { ApiRef } from '../../models';
+import {
+  DRAGEND,
+  COL_REORDER_START,
+  COL_REORDER_DRAG_OVER,
+  COL_REORDER_DRAG_OVER_HEADER,
+  COL_REORDER_DRAG_ENTER,
+  COL_REORDER_STOP,
+} from '../../constants/eventsConstants';
+import {
+  HEADER_CELL_DROP_ZONE_CSS_CLASS,
+  HEADER_CELL_DRAGGING_CSS_CLASS,
+} from '../../constants/cssClassesConstants';
+
+export interface CursorCoordinates {
+  x: number;
+  y: number;
+}
+
+const CURSOR_MOVE_DIRECTION_LEFT = 'left';
+const CURSOR_MOVE_DIRECTION_RIGHT = 'right';
+
+const reorderColDefArray = (
+  columns: ColDef[],
+  newColIndex: number,
+  oldColIndex: number,
+): ColDef[] => {
+  const columnsClone = [...columns];
+
+  columnsClone.splice(newColIndex, 0, columnsClone.splice(oldColIndex, 1)[0]);
+
+  return columnsClone;
+};
+
+const getCursorMoveDirectionX = (currentCoordinates, nextCoordinates) => {
+  return currentCoordinates.x <= nextCoordinates.x
+    ? CURSOR_MOVE_DIRECTION_RIGHT
+    : CURSOR_MOVE_DIRECTION_LEFT;
+};
+
+const hasCursorPositionChanged = (
+  currentCoordinates: CursorCoordinates,
+  nextCoordinates: CursorCoordinates,
+): boolean =>
+  currentCoordinates.x !== nextCoordinates.x || currentCoordinates.y !== nextCoordinates.y;
+
+export const useColumnReorder = (columnsRef: React.RefObject<HTMLDivElement>, apiRef: ApiRef) => {
+  const logger = useLogger('useColumnReorder');
+
+  const dragCol = React.useRef<ColDef | null>();
+  const dragColNode = React.useRef<HTMLElement | null>();
+  const cursorPosition = React.useRef<CursorCoordinates>({
+    x: 0,
+    y: 0,
+  });
+  const removeDnDStylesTimeout = React.useRef<NodeJS.Timeout | number>();
```

Comment:
It never runs on Node.js

```suggestion
  const removeDnDStylesTimeout = React.useRef<number>();
```

Reply:
I checked and the type of the `setTimeout` is `NodeJS.Timeout`, but I've removed `number` as it is not needed.

Reply:
The type should be `number`. You can disambiguate by prefixing the call with window (as not available on node).